cdist/conf/type/__filesystem/gencode-remote

```diff
@@ -0,0 +1,97 @@
+#!/bin/sh
+#
+# 2016 - 2016 Daniel Heule     (hda at sfs.biz)
+#
+# This file is part of cdist.
+#
+# cdist is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cdist is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with cdist. If not, see <http://www.gnu.org/licenses/>.
+#
+
+
+fstype="$(cat "$__object/parameter/fstype")"
+mydev="$(cat "$__object/parameter/blkdev")"
+
+label="$(cat "$__object/parameter/label")"
+mkfsoptions="$(cat "$__object/parameter/mkfsoptions")"
+
+
+if [ -f "$__object/parameter/force" ]; then
+  # create filesystem even an other filesystem is on disk or the label is not correct, use with caution !
+  forcefs="true"
+else
+  forcefs="false"
+fi
+
+
+
+blkdev_devname="$(grep -P -o2 'NAME="\K[^"]*' "$__object/explorer/lsblk")"
```

Comment:
What does the -o2 do? It seems it makes no difference if I run -o or -o2 or -o2678. Not documented in man page either.